### PR TITLE
node: pass eventbus at construction time

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: internal/consensus/state.go [hunk 1]
```diff
@@ -260,11 +262,11 @@ func (cs *State) updateStateFromStore(ctx context.Context) error {
 	return nil
 }
 
-// SetEventBus sets event bus.
-func (cs *State) SetEventBus(b *eventbus.EventBus) {
-	cs.eventBus = b
-	cs.blockExec.SetEventBus(b)
-}
+// // SetEventBus sets event bus.
```
Comment (creachadair) [inline internal/consensus/state.go:265]: Dead?

**Review by creachadair — APPROVED** [review]

**Comment by williambanfield** [comment]
Change seems great. I'm wondering if there's a need to use an actual eventbus's in place where the nop version was used before? Like tests etc?

**Comment by creachadair** [comment]
- I don't know that we use no-op event buses anywhere _except_ tests, and if that's correct I would be fine with the idea of not having the no-op implementation at all. Even if the test doesn't use the bus explicitly it's seems like constructing a real one is better unless it's obviously impractical. (The event bus doesn't have nasty dependencies for the most part, so I think that would be ok)

**Comment by tychoish** [comment]
> I don't know that we use no-op event buses anywhere _except_ tests, and if that's correct I would be fine with the idea of not having the no-op implementation at all. Even if the test doesn't use the bus explicitly it's seems like constructing a real one is better unless it's obviously impractical. (The event bus doesn't have nasty dependencies for the most part, so I think that would be ok)

I agree. I also think there's some utility in running through the production code (non-noop) code in tests assuming there isn't a big (test) performance or operations problem (e.g. I think it's fine that we use the storage engine that's just a hashmap in most places, but unless the eventbus is backing up and ooming or something, it's fine to avoid the noop case.

We definitely weren't using a noop eventbus anywhere in the prod code.

**Pull request merged**